File: android/src/main/kotlin/dev/note11/flutter_naver_map/flutter_naver_map/util/CalcUtil.kt
package dev.note11.flutter_naver_map.flutter_naver_map.util

internal object CalcUtil {
    fun float32To64(f32: Float): Double {
        val convertedInt: Int = (f32 * F32_MULTIPLIER_CONSTANT).toInt()
        return convertedInt.toDouble() / F32_MULTIPLIER_CONSTANT
    }

    private const val F32_MULTIPLIER_CONSTANT: Int = 1_000_000
}

File: ios/flutter_naver_map/Sources/flutter_naver_map/util/CalcUtil.swift
import CoreGraphics

internal struct CalcUtil {
    static func float32To64(f32: CGFloat) -> Double {
        let convertedInt: Int = Int(f32 * CGFloat(F32_MULTIPLIER_CONSTANT))
        return Double(convertedInt) / Double(F32_MULTIPLIER_CONSTANT)
    }

    private static let F32_MULTIPLIER_CONSTANT: Int = 1_000_000
}
